src/Player.cpp

```diff
@@ -27,16 +27,24 @@ void Player::draw(sf::RenderTarget &target, sf::RenderStates states) const {
   target.draw(shape, states);
 }
 
-bool Player::isGrounded() {
+bool Player::isGrounded(GameState &state) {
   sf::Vector2<float> pos = shape.getPosition();
 
-  if (pos.y < GROUND_HEIGHT)
+  sf::Vector2<float> size = shape.getSize();
+
+  bool collisionDetected = false;
+  for (int i = 0; i < shape.getSize().x; i++){
+    if (state.checkCollision(pos.x + i, pos.y + size.y)){
+      collisionDetected = true;
+      break;
+    }
+  }
+  if(!collisionDetected){
```

Comment:
```suggestion
bool Player::isGrounded(GameState &state) {
  sf::Vector2<float> pos = shape.getPosition();
  sf::Vector2<float> size = shape.getSize();

  for (int i = 0; i < shape.getSize().x; i++){
    if (state.checkCollision(pos.x + i, pos.y + size.y)){
        vy = 0;
        return true;
    }
  }
  return false;
}
```
could just return our early instead of break